Assets/Scripts/Player/PlayerStats.cs

```diff
@@ -94,16 +147,15 @@ void Update()
             }
         }
     }
-    void Regen()
+
+    void CheckRegeneration()
     {
-        if (GameSettings.gameState != GameState.InGame){return;}
-        regenTick -= Time.deltaTime;
-        if (regenTick <= 0 && regenTrue && currentHealth < maxHealth) //only works if the player is missing health
+        if (nextRegenerationTick <= 0) //Health property will deal with the max health cap
         {
-            regenTick = regenInterval;
-            CurrentHealth += regenAmount;
-            Debug.Log($"Regen: {currentHealth}");
+            nextRegenerationTick = RegenerationDelay;
+            CurrentHealth += Math.Max(((MaxHealth * RegenerationPercentage) / 100), 1); //Regenerate the % of max health per tick. Minimum 1
         }
+        nextRegenerationTick -= Time.fixedDeltaTime;
```

Comment:
Regen ticks way too fast. I tried changing fixeddeltatime here back to deltatime and that seemed to fix it.

Reply:
Fixed